packages/meditrak-server/src/tests/routes/GETAnswers.test.js

```diff
@@ -0,0 +1,181 @@
+
+    const vanuatuEntity = await findOrCreateDummyRecord(models.entity, {
+      country_code: vanuatuCountry.code,
+    });
+
+    await buildAndInsertSurveys(models, [
+      {
+        code: 'TEST_SURVEY_1',
+        name: 'Test Survey 1',
+        permission_group_id: adminPermissionGroup.id,
+        questions: [{ code: 'TEST_SURVEY_11', text: 'Best spice' }],
+      },
+      {
+        code: 'TEST_SURVEY_2',
+        name: 'Test Survey 2',
+        permission_group_id: donorPermissionGroup.id,
+        questions: [{ code: 'TEST_SURVEY_21', text: 'Worst spice' }],
+      },
+    ]);
+
+    // It's much easier to keep track of the different ids this way
+    // rather than trying to extract the correct ids from the models array
+    vanuatuAdminResponseId = generateTestId();
+    vanuatuDonorResponseId = generateTestId();
+
+    const surveyResponseModels = await buildAndInsertSurveyResponses(models, [
+      {
+        id: vanuatuAdminResponseId,
+        surveyCode: 'TEST_SURVEY_1',
+        entityCode: vanuatuEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: { TEST_SURVEY_11: 'Scary' },
+      },
+      {
+        id: vanuatuDonorResponseId,
+        surveyCode: 'TEST_SURVEY_2',
+        entityCode: vanuatuEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: { TEST_SURVEY_21: 'Sporty' },
+      },
+    ]);
+
+    for (const { surveyResponse, answers } of surveyResponseModels) {
+      if (surveyResponse.id === vanuatuAdminResponseId) {
+        vanuatuAdminAnswers = answers.map(a => a.id);
+      } else {
+        vanuatuDonorAnswers = answers.map(a => a.id);
+      }
+    }
+
+    const answerIds = vanuatuAdminAnswers.concat(vanuatuDonorAnswers);
+    filterString = `filter={"id":{"comparator":"in","comparisonValue":["${answerIds.join(
+      '","',
+    )}"]}}`;
+  });
+
+  afterEach(() => {
+    Authenticator.prototype.getAccessPolicyForUser.restore();
+  });
+
+  describe('GET /answer/:id', async () => {
+    it('Sufficient permissions: Return a requested answer if we have permission for the survey in the responses country', async () => {
+      await prepareStubAndAuthenticate(app, DEFAULT_POLICY);
+      const { body: result } = await app.get(`answer/${vanuatuAdminAnswers[0]}`);
+
+      expect(result.id).to.equal(vanuatuAdminAnswers[0]);
+    });
+
+    it('Sufficient permissions: Return a requested answer if we have BES admin access anywhere', async () => {
+      await prepareStubAndAuthenticate(app, BES_ADMIN_POLICY);
+      const { body: result } = await app.get(`answer/${vanuatuDonorAnswers[0]}`);
+
+      expect(result.id).to.equal(vanuatuDonorAnswers[0]);
+    });
+
+    it('Insufficient permissions: Throw an error if we do not have permission for the survey in the responses country for this answer', async () => {
+      await prepareStubAndAuthenticate(app, DEFAULT_POLICY);
+      const { body: result } = await app.get(`answer/${vanuatuDonorAnswers[0]}`);
```

Comment:
It all works from the perspective of the data, but having permission for Admin but not for Donor could cause confusion, given it's not a realistic situation: within our system, everyone who has Admin permission automatically gets Donor permission. Might be best to just flip every Admin to Donor and every Donor to Admin in this test suite.

Reply:
Not bothering with this